diff --git a/dist/bundled/components/story/StoryView.vue2.js b/dist/bundled/components/story/StoryView.vue2.js
index c755cf213c87017d42d5057d3e389562e74a5b6b..ec1e3b30848c56a83ef292aa08b467e8e13caec7 100644
--- a/dist/bundled/components/story/StoryView.vue2.js
+++ b/dist/bundled/components/story/StoryView.vue2.js
@@ -30,11 +30,7 @@ const _sfc_main = /* @__PURE__ */ defineComponent({
     });
     watch(() => [storyStore.currentStory, storyStore.currentVariant], () => {
       if (!storyStore.currentVariant) {
-        if (storyStore.currentStory?.lastSelectedVariant) {
-          setVariant(storyStore.currentStory.lastSelectedVariant.id);
-          return;
-        }
-        if (storyStore.currentStory?.variants.length === 1) {
+        if (storyStore.currentStory?.variants.length) {
           setVariant(storyStore.currentStory.variants[0].id);
         }
       }
diff --git a/src/app/components/story/StoryView.vue b/src/app/components/story/StoryView.vue
index 129cf2655f7ebee1114c3aa6ed4cbeb164410d04..8271c1fabde53ecd02c3f757cb05e4ff3f30a22b 100644
--- a/src/app/components/story/StoryView.vue
+++ b/src/app/components/story/StoryView.vue
@@ -28,12 +28,7 @@ watch(() => storyStore.currentVariant, (value) => {
 
 watch(() => [storyStore.currentStory, storyStore.currentVariant], () => {
   if (!storyStore.currentVariant) {
-    if (storyStore.currentStory?.lastSelectedVariant) {
-      setVariant(storyStore.currentStory.lastSelectedVariant.id)
-      return
-    }
-
-    if (storyStore.currentStory?.variants.length === 1) {
+    if (storyStore.currentStory?.variants.length) {
       setVariant(storyStore.currentStory.variants[0].id)
     }
   }
