diff --git a/dist/bundled/components/story/StoryVariantSinglePreviewRemote.vue2.js b/dist/bundled/components/story/StoryVariantSinglePreviewRemote.vue2.js
--- a/dist/bundled/components/story/StoryVariantSinglePreviewRemote.vue2.js
+++ b/dist/bundled/components/story/StoryVariantSinglePreviewRemote.vue2.js
@@ -107,7 +107,7 @@ const _sfc_main = /* @__PURE__ */ defineComponent({
             ref: iframe,
             src: sandboxUrl.value,
             class: normalizeClass(["htw-w-full htw-h-full htw-relative", {
-              "htw-invisible": !isIframeLoaded.value,
+              "htw-invisible": !isIframeLoaded.value || !__props.variant.previewReady,
               "htw-pointer-events-none": resizing
             }]),
             style: normalizeStyle(isResponsiveEnabled ? {
diff --git a/dist/bundled/components/story/StoryView.vue2.js b/dist/bundled/components/story/StoryView.vue2.js
--- a/dist/bundled/components/story/StoryView.vue2.js
+++ b/dist/bundled/components/story/StoryView.vue2.js
@@ -30,11 +30,7 @@ const _sfc_main = /* @__PURE__ */ defineComponent({
     });
     watch(() => [storyStore.currentStory, storyStore.currentVariant], () => {
       if (!storyStore.currentVariant) {
-        if (storyStore.currentStory?.lastSelectedVariant) {
-          setVariant(storyStore.currentStory.lastSelectedVariant.id);
-          return;
-        }
-        if (storyStore.currentStory?.variants.length === 1) {
+        if (storyStore.currentStory?.variants.length) {
           setVariant(storyStore.currentStory.variants[0].id);
         }
       }
diff --git a/src/app/components/story/StoryVariantSinglePreviewRemote.vue b/src/app/components/story/StoryVariantSinglePreviewRemote.vue
--- a/src/app/components/story/StoryVariantSinglePreviewRemote.vue
+++ b/src/app/components/story/StoryVariantSinglePreviewRemote.vue
@@ -129,7 +129,7 @@ function onIframeLoad() {
       :src="sandboxUrl"
       class="htw-w-full htw-h-full htw-relative"
       :class="{
-        'htw-invisible': !isIframeLoaded,
+        'htw-invisible': !isIframeLoaded || !variant.previewReady,
         'htw-pointer-events-none': resizing,
       }"
       :style="isResponsiveEnabled ? {
diff --git a/src/app/components/story/StoryView.vue b/src/app/components/story/StoryView.vue
--- a/src/app/components/story/StoryView.vue
+++ b/src/app/components/story/StoryView.vue
@@ -28,12 +28,7 @@ watch(() => storyStore.currentVariant, (value) => {
 
 watch(() => [storyStore.currentStory, storyStore.currentVariant], () => {
   if (!storyStore.currentVariant) {
-    if (storyStore.currentStory?.lastSelectedVariant) {
-      setVariant(storyStore.currentStory.lastSelectedVariant.id)
-      return
-    }
-
-    if (storyStore.currentStory?.variants.length === 1) {
+    if (storyStore.currentStory?.variants.length) {
       setVariant(storyStore.currentStory.variants[0].id)
     }
   }
